staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go

```diff
@@ -276,21 +313,29 @@ func (f *featureGate) Enabled(key Feature) bool {
 }
 
 // AddFlag adds a flag for setting global feature gates to the specified FlagSet.
-func (f *featureGate) AddFlag(fs *pflag.FlagSet) {
+func (f *featureGate) AddFlag(fs *pflag.FlagSet, component string) {
 	f.lock.Lock()
 	f.closed = true
 	f.lock.Unlock()
 
-	known := f.KnownFeatures()
+	known := f.KnownFeatures(component)
 	fs.Var(f, flagName, ""+
 		"A set of key=value pairs that describe feature gates for alpha/experimental features. "+
 		"Options are:\n"+strings.Join(known, "\n"))
 }
 
 // KnownFeatures returns a slice of strings describing the FeatureGate's known features.
-func (f *featureGate) KnownFeatures() []string {
+// If list of componenets are specified, only feature applies to those components will be returned.
+func (f *featureGate) KnownFeatures(component string) []string {
 	var known []string
-	for k, v := range f.known.Load().(map[Feature]FeatureSpec) {
+	allFeatures := f.known.Load().(map[Feature]FeatureSpec)
+	selectedFeatures := make(map[Feature]FeatureSpec)
+	for k, v := range allFeatures {
+		if v.hasComponent(component) {
+			selectedFeatures[k] = v
+		}
+	}
+	for k, v := range selectedFeatures {
```

Comment:
to avoid iterating twice and using 2 structs, you can do

```
for k, v := range allFeatures {
   if !v.hasComponent(component) {
      continue
   }
   ...
```

Reply:
Fixed.